lib/browser/api/web-contents.ts

```diff
@@ -493,6 +493,12 @@ const addReplyInternalToEvent = (event: any) => {
   });
 };
 
+const addSenderFrameToEvent = (event: any) => {
+  Object.defineProperty(event, 'senderFrame', {
+    get: () => webFrameMain.fromId(event.processId, event.frameId)
+  });
```

Comment:
I'd prefer this to be resistant to the following:

```js
wc.on('event', (event) => {
  event.processId = 123
  event.senderFrame // ==> returns some other frame because the process ID changed
})
```

```suggestion
  const { processId, frameId } = event
  Object.defineProperty(event, 'senderFrame', {
    get: () => webFrameMain.fromId(processId, frameId)
  });
```